rust/src/encodings/plain.rs

```diff
@@ -243,13 +244,37 @@ impl<'a> PlainDecoder<'a> {
     }
 
     async fn take_boolean(&self, indices: &UInt32Array) -> Result<ArrayRef> {
-        // TODO: optimize boolean access
-        let start = indices.value(0) as usize;
-        let end = indices.value(indices.len() - 1) as usize;
-        let array = self.get(start..end + 1).await?;
-        let array_byte_boundray = (start / 8 * 8) as u32;
-        let shifted_indices = subtract_scalar(indices, array_byte_boundray)?;
-        Ok(take(array.as_ref(), &shifted_indices, None)?)
+        let block_size = self.reader.prefetch_size() as u32;
+
+        let mut chunk_ranges = vec![];
```

Comment:
Lets' make these chunks aligned with 4K or 16k block size?

Reply:
Updated like this
````
            if (indices.value(j as usize + 1) / boolean_block_size) >
                (indices.value(start as usize) / boolean_block_size) {
```

Reply:
Shall we update the [code for primitive types](https://github.com/eto-ai/lance/blob/0763604a5a6e477d6ecdced860b10b9428813073/rust/src/encodings/plain.rs#L305-L306) as well?

Reply:
primtive types seems fine. Let's focus on boolean now.